contentcuration/contentcuration/viewsets/channel.py

```diff
@@ -484,6 +485,15 @@ def sync(self, request, pk=None):
         task, task_info = create_async_task("sync-channel", request.user, **task_args)
         return Response("")
 
+    def update_from_changes(self, changes):
+        errors, changes_to_return = super(ChannelViewSet, self).update_from_changes(changes)
+
+        # Recalculate storage if channel is being deleted or restored
+        if any([c for c in changes if 'deleted' in c['mods']]):
+            changes_to_return.append({CALCULATE_STORAGE_FLAG: True})
```

Comment:
Note that these are not valid 'changes', see the `generate_update_event` function referenced in this file for an example of generating one.

I also think it is probably just easier to initiate the file size recalculation task here, and rely on the frontend polling to pick that up when it has completed. So, we can probably dispense with these extra changes, and just call your `calculate_user_storage` function from `contentcuration.utils.user` instead.